aspnetcore/blazor/performance/webassembly-browser-developer-tools-diagnostics.md

```diff
@@ -0,0 +1,202 @@
+
+## Counters trace
+
+* Manual testing
+  * Browser developer tools: Download the `.json` output file, open the file in Visual Studio, and find the expected counters.
+  * [`dotnet-counters collect`](/dotnet/core/diagnostics/dotnet-counters): Open the `.csv`/`.json` output file in Visual Studio and find the expected counters.
+* Web-based testing
+  * Upload the file via HTTP.
+  * Parse and validate that the trace contains the expected counters.
+
+## .NET Core Diagnostics Client Library example
+
+Parse and validate NetTrace (`.nettrace`) messages using the .NET Core Diagnostics Client Library:
+
+* [`dotnet/diagnostics` GitHub repository](https://github.com/dotnet/diagnostics)
+* [`Microsoft.Diagnostics.NETCore.Client` NuGet package](https://www.nuget.org/packages/Microsoft.Diagnostics.NETCore.Client)
+
+For more information, see the [.NET Core diagnostics documentation](/dotnet/core/diagnostics/) and the [`IpcMessage` API (reference source)](https://github.com/dotnet/diagnostics/blob/main/src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsIpc/IpcMessage.cs).
+
+[!INCLUDE[](~/includes/aspnetcore-repo-ref-source-links.md)]
+
+The following example:
+
+* Collects a GC (Garbage Collector) dump of the live .NET process.
+* Collects performance counters for 60 seconds.
+* Collects CPU counters for 60 seconds.
+
+In the project file (`.csproj`), the properties in the following table enable integration with the browser's profiler.
+
+Property | Default | Set value to&hellip; | Description
+--- | :---: | :---: | ---
+`<WasmProfilers>` | No value | `browser` | Mono profilers to use. Potential values are "`browser`" and "`log`". To use both, separate the values with a semicolon. The `browser` profiler enables integration with the browser's developer tools profiler.
+`<WasmPerfTracing>` | `false` | `true` | Controls diagnostic server tracing.
+`<WasmPerfInstrumentation>` | `false` | `true` | Controls CPU sampling instrumentation for diagnostic server.
+`<MetricsSupport>` | `false` | `true` | Controls `System.Diagnostics.Metrics` support. For more information, see the [`System.Diagnostics.Metrics` namespace](/dotnet/api/system.diagnostics.metrics).
+`<EventSourceSupport>` | `false`| `true` | Controls `EventPipe` support. For more information, see [Diagnostics and instrumentation: Observability and telemetry](/dotnet/core/deploying/native-aot/diagnostics#observability-and-telemetry).
+
+```xml
+<PropertyGroup>
+  <WasmProfilers>browser</WasmProfilers>
+  <WasmPerfTracing>true</WasmPerfTracing>
+  <WasmPerfInstrumentation>true</WasmPerfInstrumentation>
+  <MetricsSupport>true</MetricsSupport>
+  <EventSourceSupport>true</EventSourceSupport>
+</PropertyGroup>
+```
+
+The [`Timing-Allow-Origin` HTTP header](https://developer.mozilla.org/docs/Web/HTTP/Reference/Headers/Timing-Allow-Origin) allows for more precise time measurements.
+
+Browser developer tools console calls in the following example that trigger profiling:
```

Comment:
Sorry, this must be confusing. Those commands do not belong to this page, but to EP page.

globalThis.getDotnetRuntime(0).collectGcDump();
globalThis.getDotnetRuntime(0).collectPerfCounters({durationSeconds: 60});
globalThis.getDotnetRuntime(0).collectCpuSamples({durationSeconds: 60});